service/worker/service.go

```diff
@@ -266,37 +270,14 @@ func (s *Service) startIndexer() {
 }
 
 func (s *Service) startArchiver() {
-	historyArchiverBootstrapContainer := &carchiver.HistoryBootstrapContainer{
-		HistoryV2Manager: s.GetHistoryManager(),
-		Logger:           s.GetLogger(),
-		MetricsClient:    s.GetMetricsClient(),
-		ClusterMetadata:  s.GetClusterMetadata(),
-		DomainCache:      s.GetDomainCache(),
-	}
-	visibilityArchiverBootstrapContainer := &carchiver.VisibilityBootstrapContainer{
-		Logger:          s.GetLogger(),
-		MetricsClient:   s.GetMetricsClient(),
-		ClusterMetadata: s.GetClusterMetadata(),
-		DomainCache:     s.GetDomainCache(),
-	}
-	archiverProvider := s.GetArchiverProvider()
-	err := archiverProvider.RegisterBootstrapContainer(
-		common.WorkerServiceName,
-		historyArchiverBootstrapContainer,
-		visibilityArchiverBootstrapContainer,
-	)
-	if err != nil {
-		s.GetLogger().Fatal("failed to register archiver bootstrap container", tag.Error(err))
-	}
-
```

Comment:
if these logic is archival worker specific, then plz keep it here
resource is for common use case (frontend / history / matching / worker)

Reply:
It's common logic for frontend, history and worker, so moved to resource.